src/controllers/aer_controller.hpp

```diff
@@ -917,18 +565,7 @@ Result Controller::execute(std::vector<Circuit> &circuits,
 
```

Comment:
I think `Controller::execute(std::vector<Circuit> &circuits, ..)` method can be

1. Determine methods (calling `simulation_methods()`)
2. Construct `std::vector<std::shared_ptr<Executor>>`
3. Determine experiment parallelization (`set_parallelization_experiments()`) by using `Executor.required_memory_mb()`
4. Call `Executor. run_circuit()` in parallel or serial

Construction of `Executor` can be via a static method in `simulators.hpp`.

Reply:
I create `Controller::make_circuit_executor` function and removed `required_memory_mb` and `validate_state` from `Controller`.